### PR TITLE
Warning if no background

### LINKED ISSUE #214: Better Warnings for Objects without Background
**Is your feature request related to a problem? Please explain.**

It is hard to sometimes keep track of an Action being defined outside of the frames for a given animation. Especially in the case of defined BackgroundActions. Without a BackgroundAction, the Action will not render properly (if at all) and there ought to be  warning a user to this possible problem.

**Describe the solution you'd like**

I would like to add the ability to alert a user that a BackgroundAction frame is not present for a given Action that is defined outside the range provided by the BackgroundAction. A user could potentially define multiple different backgrounds so the solution will have to account for this.

### PR DESCRIPTION
**Pull request by Wikunia**
## PR Checklist

If you are contributing to `Javis.jl`, please make sure you are able to check off each item on this list:

- [x] Did I update `CHANGELOG.md` with whatever changes/features I added with this PR?
- [x] Did I make sure to only change the part of the file where I introduced a new change/feature?
- [x] Did I cover all corner cases to be close to 100% test coverage (if applicable)?
- [x] Did I properly add Javis dependencies to the `Project.toml` + set an upper bound of the dependency (if applicable)?
- [x] Did I properly add test dependencies to the `test` directory (if applicable)?
- [x] Did I check relevant tutorials that may be affected by changes in this PR?
- [x] Did I clearly articulate why this PR was made the way it was and how it was made?

**Link to relevant issue(s)**
Closes #214 

**How did you address these issues with this PR? What methods did you use?**
The video now saves a range for which a background is defined. The `preprocess_frames!` function then checks whether some frames don't have a background and throws a warning. 
Additionally I've changed some test cases to not throw a warning :smile:

### CODE REVIEW

**Comment by Wikunia** [comment]
I have no idea why format is failing. I've run the `format` command. Maybe you can double check this @TheCedarPrince

File: src/Javis.jl [hunk 1]
```diff
@@ -144,6 +147,10 @@ function preprocess_frames!(objects::Vector{<:AbstractObject})
         append!(frames, collect(get_frames(object)))
     end
     frames = unique(frames)
+    if !(frames ⊆ CURRENT_VIDEO[1].background_frames)
+        @warn("Some of the frames don't have a background: In this case: ",
```
Comment (TheCedarPrince) [inline src/Javis.jl:151]: "background:" to "background."

File: src/Javis.jl [hunk 2]
```diff
@@ -144,6 +147,10 @@ function preprocess_frames!(objects::Vector{<:AbstractObject})
...
+             setdiff(frames, CURRENT_VIDEO[1].background_frames))
+    end
```
Comment (TheCedarPrince) [inline src/Javis.jl:153]: I feel that the warning is a bit verbose:

```
┌ Warning: Some of the frames don't have a background: In this case:
│   setdiff(frames, (CURRENT_VIDEO[1]).background_frames) =
│    1-element Array{Int64,1}:
│     11
└ @ Javis /home/src/Projects/javis/src/Javis.jl:151
```

Perhaps instead we could do this: 

```julia
        @warn("Some of the frames don't have a background: In this case: $(
             setdiff(frames, CURRENT_VIDEO[1].background_frames))")
```

which I think produces the nicer warning:

```
┌ Warning: Some of the frames don't have a background: In this case: [11]
└ @ Javis /home/src/Projects/javis/src/Javis.jl:151
```

**Review by TheCedarPrince — CHANGES_REQUESTED** [review]
This makes sense - I found one typo and made a minor suggestion on implementation. Only other question that I had is should we seek to identify which Object does not have a background defined?

**Comment by Wikunia** [comment]
> Only other question that I had is should we seek to identify which Object does not have a background defined?
I don't think we can.

**Review by TheCedarPrince — APPROVED** [review]
